registry/coder/modules/vscode-desktop-core/main.tf

```diff
@@ -0,0 +1,90 @@
+terraform {
+  required_version = ">= 1.0"
+
+  required_providers {
+    coder = {
+      source  = "coder/coder"
+      version = ">= 2.5"
+    }
+  }
+}
+
+variable "agent_id" {
+  type        = string
+  description = "The ID of a Coder agent."
+}
+
+variable "folder" {
+  type        = string
+  description = "The folder to open in the IDE."
+  default     = ""
+}
+
+variable "open_recent" {
+  type        = bool
+  description = "Open the most recent workspace or folder. Falls back to the folder if there is no recent workspace or folder to open."
+  default     = false
+}
+
+variable "protocol" {
```

Comment:
we have a white list of protocols in `coder/coder` to inject the secret `SESSION_TOKEN`. So, maybe we add a validation here to control what we support.

Reply:
hmm i'll find that list

Reply:
It wasn't asking when I was doing the kiro module.

Reply:
But yeah we can skip validation

Reply:
I will just add a note then